score-lang-compiler/src/main/java/org/openscore/lang/compiler/scorecompiler/ScoreCompilerImpl.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * (c) Copyright 2014 Hewlett-Packard Development Company, L.P.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License v2.0 which accompany this distribution.
+ *
+ * The Apache License is available at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.openscore.lang.compiler.scorecompiler;
+
+import ch.lambdaj.function.convert.Converter;
+import org.apache.commons.collections4.MapUtils;
+import org.apache.commons.lang.Validate;
+import org.openscore.api.ExecutionPlan;
+import org.openscore.lang.compiler.SlangTextualKeys;
+import org.openscore.lang.compiler.modeller.DependenciesHelper;
+import org.openscore.lang.compiler.modeller.model.Executable;
+import org.openscore.lang.compiler.modeller.model.Flow;
+import org.openscore.lang.compiler.modeller.model.Operation;
+import org.openscore.lang.entities.CompilationArtifact;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static ch.lambdaj.Lambda.convertMap;
+
+/*
+ * Created by stoneo on 2/2/2015.
+ */
+
+@Component
+public class ScoreCompilerImpl implements ScoreCompiler{
+
+    @Autowired
+    private ExecutionPlanBuilder executionPlanBuilder;
+
+    @Autowired
+    private DependenciesHelper dependenciesHelper;
+
+    @Override
+    public CompilationArtifact compile(Executable executable, Set<Executable> path) {
+
+        Map<String, Executable> filteredDependencies = new HashMap<>();
+        //we handle dependencies only if the file has imports
+        boolean hasDependencies = MapUtils.isNotEmpty(executable.getDependencies())
```

Comment:
System properties are currently not considered as dependencies at all.
We need to rethink it once we decide how to treat system props in general